apps/aecore/src/aec_headers.erl

```diff
@@ -41,14 +41,35 @@ get_by_hash(_Hash) ->
 -spec serialize_for_network(header()) ->
                                    {ok, block_header_serialized_for_network()}.
 serialize_for_network(H = #header{}) ->
-    %% TODO: Define serialization format.
-    {ok, term_to_binary(H)}.
+    Serialized =
+      #{<<"height">> =>  height(H),
+        <<"prev-hash">> => prev_hash(H),
+        <<"root-hash">> => H#header.root_hash,
+        <<"difficulty">> => H#header.difficulty,
+        <<"nonce">> => H#header.nonce,
+        <<"time">> => H#header.time,
+        <<"version">> => H#header.version
+      },
+    {ok, Serialized}.
```

Comment:
I did a quick search if we can't do it automatically. It should be possible to do some macro, but I haven't found anything and this isn't too messy.